statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,436 @@
+    """
+
+    theta_hat = np.eye(p)
+    for pi in range(p):
+        ind = list(range(p))
+        ind.pop(pi)
+        theta_hat[pi,ind] = gamma_hat_l[pi]
+        theta_hat[pi,:] = (- 1. / tau_hat_l[pi]**2) * theta_hat[pi,:]
+
+    return theta_hat
+
+
+def _est_regularized_distributed(mod, mnum, partitions, fit_kwds=None,
+                                 score_kwds=None, hess_kwds=None):
+    """generates the regularized fitted parameters, is the default
+    estimation_method for distributed_estimation
+
+    Parameters
+    ----------
+    mod : statsmodels model class
+        The model for the current machine.
+    mnum : scalar
+        Index of current machine
+    partitions : scalar
+        Total number of machines
+    fit_kwds : dict-like or None
+        Keyword arguments to be given to fit_regularized
+    score_kwds : dict-like or None
+        Keyword arguments for the score function.
+    hess_kwds : dict-like or None
+        Keyword arguments for the Hessian function.
+
+    Returns
+    -------
+    A tuple of paramters for regularized fit
+        An array-like object of the fitted parameters, beta hat
+        An array-like object for the gradient
+        A list of array like objects for gamma hat
+        A list of array like objects for tau hat
+    """
+
+    score_kwds = {} if score_kwds is None else score_kwds
+    hess_kwds = {} if hess_kwds is None else hess_kwds
+
+    if fit_kwds is None:
+        raise ValueError("_est_regularized_distributed currently " +
+                         "requires that fit_kwds not be None.")
+    else:
+        alpha = fit_kwds["alpha"]
+
+    if "L1_wt" in fit_kwds:
+        L1_wt = fit_kwds["L1_wt"]
+    else:
+        L1_wt = 1
+
+    n_obs, p = mod.exog.shape
+    p_part = int(np.ceil((1. * p) / partitions))
+
+    params = mod.fit_regularized(**fit_kwds).params
+    grad = _gen_grad(mod, params, alpha, L1_wt, score_kwds) / n_obs
+
+    X_beta = _gen_wdesign_mat(mod, params, hess_kwds)
+
+    gamma_hat_l = []
+    tau_hat_l = []
+    for pi in range(mnum * p_part, min((mnum + 1) * p_part, p)):
+
+        gamma_hat = _gen_gamma_hat(X_beta, pi, alpha)
+        gamma_hat_l.append(gamma_hat)
+
+        tau_hat = _gen_tau_hat(X_beta, gamma_hat, pi, alpha)
+        tau_hat_l.append(tau_hat)
+
+    return params, grad, gamma_hat_l, tau_hat_l
+
+
+def _join_debiased(model_results_l, partitions, threshold=0):
+    """joins the results from each run of _est_regularized_distributed
+    and returns the debiased estimate of the coefficients
+
+    Parameters
+    ----------
+    model_results_l : list
+        A list of tuples each one containing the beta_hat, grad,
+        gamma_hat and tau_hat values for each partition.
+    partitions : scalar
+        The number of partitions that the data will be split into.
+    threshold : scalar
+        The threshold at which the coefficients will be cut.
+    """
+
+    params_l = []
```

Comment:
Since params_l and grad_l are only used to create params_mn and grad_mn, it's better to create the means directly rather than having a list as an intermediate.